### ISSUE TITLE
Missing argument validation in AnyCallConfiguration.WhenArgumentsMatch

### ISSUE THREAD

**Issue by thomaslevesque** [FakeItEasy/FakeItEasy#1370]
This code throws `NullReferenceException` when the fake method is called. It should throw `ArgumentNullException` when `WhenArgumentsMatch` instead.

```c#
var foo = A.Fake<IFoo>();
A.CallTo(foo).WhenArgumentsMatch(null).DoesNothing();
foo.Bar();
```

**Comment by thomaslevesque** [comment]
`Where` exhibits the same problem (but is less easily encountered, because `WhereConfigurationExtensions.Where`, which is used more frequently, does check the argument)

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy 4.7.0](https://github.com/FakeItEasy/FakeItEasy/releases/4.7.0).
